src/graph/backend/elyzor/target_machine.hpp

```diff
@@ -16,35 +16,26 @@
 #ifndef BACKEND_ELYZOR_TARGET_MACHINE_HPP
 #define BACKEND_ELYZOR_TARGET_MACHINE_HPP
 
-// An internal GC's functionality that provides machine flags,
-// assuming all the flags are true for now
-// #include "core/src/compiler/config/context.hpp"
-// #include "runtime/config.hpp"
+#include "cpu/x64/cpu_isa_traits.hpp"
 
 #define REQUIRE_AVX512_BEGIN \
-    if (true) {
-    // if (::dnnl::impl::graph::gc::get_default_context() \
-    //                 ->machine_.cpu_flags_.fAVX512F) {
+    if (dnnl::impl::cpu::x64::cpu().has(Xbyak::util::Cpu::tAVX512F)) {
 #define REQUIRE_VNNI_AMXINT8_BEGIN \
-    if (true) {
-    // if (::dnnl::impl::graph::gc::get_default_context() \
-    //                 ->machine_.cpu_flags_.fAVX512VNNI \
-    //         || ::dnnl::impl::graph::gc::get_default_context() \
-    //                    ->machine_.cpu_flags_.fAVX512AMXINT8) {
+    if (dnnl::impl::cpu::x64::cpu().has(Xbyak::util::Cpu::tAVX512_VNNI) \
+            || (dnnl::impl::cpu::x64::cpu().has(Xbyak::util::Cpu::tAVX512F) \
+                    && dnnl::impl::cpu::x64::cpu().has( \
+                            Xbyak::util::Cpu::tAMX_INT8))) {
```

Comment:
gc internals:
- AVX512VNNI: [has_avx512 && check 11th bit of info[2]](https://github.com/oneapi-src/oneDNN/blob/1344a14b4160d62a18dcc447a2c0f122f87b5b03/src/graph/backend/graph_compiler/core/src/runtime/target_machine.cpp#L166-L167)
- AVX512AMXINT8: [has_avx512 && check 25th bit of info[3]](https://github.com/oneapi-src/oneDNN/blob/1344a14b4160d62a18dcc447a2c0f122f87b5b03/src/graph/backend/graph_compiler/core/src/runtime/target_machine.cpp#L174-L175)

Xbyak utils:
- tAVX512_VNNI: [check 11th bit of ECX](https://github.com/oneapi-src/oneDNN/blob/1344a14b4160d62a18dcc447a2c0f122f87b5b03/src/cpu/x64/xbyak/xbyak_util.h#L630) (same as info[2]) [if avx512](https://github.com/oneapi-src/oneDNN/blob/1344a14b4160d62a18dcc447a2c0f122f87b5b03/src/cpu/x64/xbyak/xbyak_util.h#L620)
- tAMX_INT8: [check 25th bit of EDX (same as info[3])](https://github.com/oneapi-src/oneDNN/blob/1344a14b4160d62a18dcc447a2c0f122f87b5b03/src/cpu/x64/xbyak/xbyak_util.h#L670)